python/lsst/ts/Dome/mock_llc_statuses/mock_motion/azimuth_motion.py

```diff
@@ -0,0 +1,234 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AzimuthMotion"]
+
+import logging
+import math
+
+from lsst.ts.idl.enums.Dome import MotionState
+
+
+class AzimuthMotion:
+    """Simulator for the azimuth motion of the Dome.
+
+    Parameters
+    ----------
+    initial_position: `float`
+        The initial position [rad].
+    max_speed: `float`
+        The maximum allowed speed [rad/s].
+    current_tai: `float`
+        The current TAI time.
+
+    Notes
+    -----
+    This simulator can either move the dome to a target position at maximum
+    speed and start crawling from there with the specified crawl velocity, or
+    crawl at the specified velocity. It handles the 0/2pi radians (0/360
+    degrees) boundary. When either moving or crawling, a new move or handle
+    command is handled, the azimuth motion/crawl can be stopped and the dome
+    can be parked.
+    """
+
+    def __init__(self, initial_position, max_speed, current_tai):
+        self._initial_position = initial_position
+        self._motion_state = MotionState.STOPPED
+        self._commanded_tai = current_tai
+        self._target_position = 0
+        self._max_speed = max_speed
+        self._move_velocity = 0
+        self._crawl_velocity = 0
+        self._do_move = True
+        self.log = logging.getLogger("MockCircularCrawlingActuator")
+
+    def determine_distance_and_move_speed(self):
+        """Determines the smallest distance [rad] between the initial and
+        target positions assuming motion around a circle. The move velocity is
+        determined at the same time as well.
+
+        Returns
+        -------
+        distance: `float`
+            The smallest distance between the initial and target positions.
+
+        """
+        distance = self._target_position - self._initial_position
+        self._move_velocity = self._max_speed
+        if distance >= math.pi:
+            distance -= 2 * math.pi
+            self._move_velocity = -self._max_speed
+        elif distance > 0:
+            self._move_velocity = self._max_speed
+        elif distance < -math.pi:
+            distance += 2 * math.pi
+            self._move_velocity = self._max_speed
+        elif distance < 0:
+            self._move_velocity = -self._max_speed
+        return distance
```

Comment:
Thanks very much for this suggestion. I remembered that I saw this somewhere but I forgot where, so thanks for the reminder. Yes, EIE insist on using rad and not deg. I will test the salobj code to see if it works for me. Since that code assumes deg I wonder if the code becomes clearer if I simply copy the methods into my code and use u.rad instead of u.deg. Perhaps we can discuss adding a rad version of that code in salobj.